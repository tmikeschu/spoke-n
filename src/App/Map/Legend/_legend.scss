.legend {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px dotted $yellow;
  color: $green;
  img {
    width: 7%;
  }
  h4 {
    width: 50%;
    text-align: center;
    margin: 2% auto;
  }
  ul {
    margin: 5% auto;
    width: 50%;
    li {
      margin: 10px 0;
    }
  }

  @media (max-width: 640px) {
    flex-direction: column;
    h4 {
      font-size: 1em;
      width: 100%;
    }
    ul {
      width: 100%;
      li {
        font-size: .5em;
        margin-left: 5%;
      }
    }
  }
}
