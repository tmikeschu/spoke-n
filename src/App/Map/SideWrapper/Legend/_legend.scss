.legend {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border: 1px dotted $yellow;
  color: $green;
  img {
    width: 7%;
  }
  h4 {
    margin: 2%;
  }
  ul {
    margin: 2%;
    li {
      margin: 10px 0;
      &:nth-of-type(3) {
        span {
          color: #FB7064;
        }
      }
      img[alt="flag"] {
        transition: 300ms;
        &:hover, &:focus {
          transition: 300ms;
          transform: scale(1.1);
          cursor: pointer;
        }
      }
    }
  }

  @media (max-width: 640px) {
    h4 {
      font-size: 1em;
      width: 100%;
    }
    ul {
      width: 100%;
      li {
        font-size: .5em;
        margin-left: 2%;
        &:nth-of-type(3) {
          span {
            color: #FB7064;
            margin-left: 0;
            margin-right: 0;
          }
        }
      }
    }
  }
}
