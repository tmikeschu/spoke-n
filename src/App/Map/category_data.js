const categoryIcons = {
  stay: "https://maps.google.com/mapfiles/ms/icons/blue.png",
  checkout: "https://maps.google.com/mapfiles/ms/icons/green.png",
  avoid: "https://maps.google.com/mapfiles/ms/icons/red.png",
  bike_shop: "https://maps.google.com/mapfiles/ms/icons/mechanic.png",
  other: "https://maps.google.com/mapfiles/ms/icons/purple.png",
  endOfDay: "https://maps.google.com/mapfiles/ms/icons/flag.png",
  cycling: "https://maps.google.com/mapfiles/ms/icons/cycling.png",
}

const categoryLabels = {
  "stay": "Places to Stay",
  "checkout": "Cool Spots",
  "avoid": "Places to Avoid",
  "bike_shop": "Bike Shops",
  "other": "Other",
  "": "All",
  "DISPLAYNONE": "None"
};

export { categoryIcons, categoryLabels }

