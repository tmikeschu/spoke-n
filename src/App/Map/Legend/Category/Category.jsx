import React, { Component } from 'react';
import { categoryIcons, categoryLabels } from '../../category_data';

export default class Category extends Component {
  categoryIcons() {
    return categoryIcons;
  }

  categoryLabels() {
    return categoryLabels;
  }

  render() {
    const category = this.props.category;

    return(
      <li style={{display: this.props.tabIndex === 0 ? 'none' : 'block'}}>
        <span>
          <img 
            src={this.categoryIcons()[category]} 
            alt={category}
          />
        </span> 
        â†’ { this.categoryLabels()[category] }
      </li>
    );
  }
}
