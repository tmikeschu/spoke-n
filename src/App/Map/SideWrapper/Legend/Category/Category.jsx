import React from "react"
import PropTypes from "prop-types"
import Emoji from "../../../Emoji"
import { categoryIcons, categoryLabels } from "../../../category_data"

const Category = ({ category }) => (
  <li>
    <Emoji emoji={categoryIcons[category]} />
    â†’ {categoryLabels[category]}
  </li>
)

Category.propTypes = {
  category: PropTypes.string.isRequired,
}

export default Category
