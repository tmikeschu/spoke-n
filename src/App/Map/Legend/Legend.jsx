import React, { Component } from 'react';
import Category from './Category/Category';
import '../../App.css'

export default class Legend extends Component {
  render() {
    const legendCategories = this.props.categories.map((category, i) => {
      return(
        <Category key={i} category={category} />
      );
    });

    return (
      <article className="legend">
        <h4>Legend</h4>
        <ul>
          <li><span style={{fontWeight: "bold"}}>―― </span> → Planned Route</li>
          <li><span style={{fontWeight: "bold", color: "#f00"}}>―― </span> → Actual Route</li>
          <li>
            <span>
              <img src="http://maps.google.com/mapfiles/ms/icons/cycling.png" alt="cyclist"/>
            </span> → That's Us! ( as of {this.props.date} )
          </li>
          { legendCategories }
        </ul>
      </article>
    );
  }
}
