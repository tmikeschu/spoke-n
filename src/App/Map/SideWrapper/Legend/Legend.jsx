import React from 'react'
import PropTypes from 'prop-types'
import Category from './Category/Category'
import { categoryIcons } from '../../category_data'

const Legend = ({ categories, date, toggleFlags }) => {
  const legendCategories = categories.map((category, i) => ( 
    <Category key={i} category={category} />
  ))

  return (
    <article className="legend">
      <h4>Legend</h4>
      <ul>
        <li><span style={{fontWeight: "bold"}}>―― </span> → Planned Route</li>
        <li><span style={{fontWeight: "bold", color: "#f00"}}>―― </span> → Actual Route</li>
        <li>
          <span>
            <img src={categoryIcons["cycling"]} alt="cyclist"/>
          </span> → That's Us! ( as of {date} )
        </li>
        <li>
          <span>
            <img src={categoryIcons["endOfDay"]} alt="flag" onClick={toggleFlags} />
          </span> → Day-to-Day (click flag to hide/show flag markers)
        </li>
        { legendCategories }
      </ul>
    </article>
  )
}

Legend.defaultProps = {
  date: "",
  categories: []
}

Legend.propTypes = {
  categories: PropTypes.array.isRequired,
  date: PropTypes.string.isRequired
}

export default Legend
