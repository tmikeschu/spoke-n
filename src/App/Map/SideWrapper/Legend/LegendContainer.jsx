import { connect } from "react-redux"
import { bindActionCreators } from "redux"
import * as actions from "../../../../Redux/actions"
import Legend from "./Legend"

const mapStateToProps = state => ({
  date: state.map.currentLocation.date
})

const mapDispatchToProps = dispatch => ({
  toggleFlags: bindActionCreators(actions, dispatch).toggleFlags
})

export default connect(mapStateToProps, mapDispatchToProps)(Legend)

