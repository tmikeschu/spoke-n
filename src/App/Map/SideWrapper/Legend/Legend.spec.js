import React from "react"
import { shallow } from "enzyme"
import Legend from "./Legend"
import Category from './Category/Category'
import { categoryIcons } from '../../category_data'

describe("<Legend />", () => {
  const wrapper = shallow(<Legend categories={["stay"]} date="TODAY" />)

  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
