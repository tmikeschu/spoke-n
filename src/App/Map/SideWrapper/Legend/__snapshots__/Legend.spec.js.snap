// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Legend /> snapshot is valid 1`] = `
Array [
  <article
    className="legend"
>
    <h4>
        Legend
    </h4>
    <ul>
        <li>
            <span
                style={
                    Object {
                        "fontWeight": "bold",
                      }
                }
            >
                ―― 
            </span>
             → Planned Route
        </li>
        <li>
            <span
                style={
                    Object {
                        "color": "#f00",
                        "fontWeight": "bold",
                      }
                }
            >
                ―― 
            </span>
             → Actual Route
        </li>
        <li>
            <span>
                <img
                    alt="cyclist"
                    src="https://maps.google.com/mapfiles/ms/icons/cycling.png"
                />
            </span>
             → That's Us! ( as of 
            TODAY
             )
        </li>
        <li>
            <span>
                <img
                    alt="flag"
                    onClick={undefined}
                    src="https://maps.google.com/mapfiles/ms/icons/flag.png"
                />
            </span>
             → Day-to-Day (click flag to hide/show flag markers)
        </li>
        <Category
            category="stay"
        />
    </ul>
</article>,
]
`;
